apicapi/apic_manager.py

```diff
@@ -301,14 +300,19 @@ def ensure_entity_profile_created_on_apic(
             self, name, domain_dn=None, enable_infra_vlan=True,
             incl_vmware_vmm=True, transaction=None):
         """Create the infrastructure entity profile."""
-        if not self.provision_infra:
-            return
 
         with self.apic.transaction(transaction) as trs:
-            self.apic.infraAttEntityP.create(name, transaction=trs)
-            if enable_infra_vlan:
-                self.apic.infraProvAcc.create(name, transaction=trs)
-            # Attach domain(s) to entity profile
+            if self.provision_infra:
+                self.apic.infraAttEntityP.create(name, transaction=trs)
+                if enable_infra_vlan:
+                    self.apic.infraProvAcc.create(name, transaction=trs)
+
+            # Always attach domain(s) to entity profile if AEP does exist
+            if (not self.provision_infra and
+                self.apic.infraAttEntityP.get(name) is None):
```

Comment:
move this operation outside the apic transaction, so we can make sure that it's executed after anything that is in it.